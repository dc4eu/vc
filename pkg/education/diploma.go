package education

import "encoding/json"

type DiplomaDocument struct {
	DataOfBirth string `json:"date_of_birth" validate:"required"`
	FamilyName  string `json:"family_name" validate:"required"`
	GivenName   string `json:"given_name" validate:"required"`

	PersonalIdentifier PersonalIdentifier `json:"personal_identifier"`

	NameOfAwardingTertiaryEducationInstitution string `json:"name_of_awarding_tertiary_education_institution" validate:"required"`

	NameOfQualification                             string `json:"name_of_qualification" validate:"required"`
	DateOfAward                                     string `json:"date_of_award" validate:"required"`
	CountryOfAwardOfAcademicQualification           string `json:"country_of_award_of_academic_qualification"`
	OverallClassificationOfTheAcademicQualification string `json:"overall_classification_of_the_academic_qualification" validate:"required"`
	NameOfQualificationStudyField                   string `json:"name_of_qualification_study_field"`
	DegreeProjectTitle                              string `json:"degree_project_title"`
	Entitlement                                     string `json:"entitlement"`
	OtherInformation                                string `json:"other_information"`
}

func (d *DiplomaDocument) Marshal() (map[string]any, error) {
	data, err := json.Marshal(d)
	if err != nil {
		return nil, err
	}

	var doc map[string]any
	err = json.Unmarshal(data, &doc)
	if err != nil {
		return nil, err
	}

	return doc, nil
}

type PersonalIdentifier struct {
	NationalIDNumber string `json:"national_id_number"`
	Citizenship      string `json:"citizenship"`
	DateOfBirth      string `json:"date_of_birth"`
	Gender           string `json:"gender"`
	GroupMemberOf    string `json:"group_member_of"`
	HasClaim         string `json:"has_claim"`
	HasCredential    string `json:"has_credential"`
	PlaceOfBirth     string `json:"place_of_birth"`
}
