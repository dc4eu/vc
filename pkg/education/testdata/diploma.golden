{
  "country_of_award_of_academic_qualification": "test-country-of-award-of-academic-qualification",
  "date_of_award": "test-date-of-award",
  "date_of_birth": "test-date-of-birth",
  "degree_project_title": "test-degree-project-title",
  "entitlement": "test-entitlement",
  "family_name": "test-family-name",
  "given_name": "test-given-name",
  "name_of_awarding_tertiary_education_institution": "test-name-of-awarding-tertiary-education-institution",
  "name_of_qualification": "test-name-of-qualification",
  "name_of_qualification_study_field": "test-name-of-qualification-study-field",
  "other_information": "test-other-information",
  "overall_classification_of_the_academic_qualification": "test-overall-classification-of-the-academic-qualification",
  "personal_identifier": {
    "citizenship": "test-citizenship",
    "date_of_birth": "test-date-of-birth",
    "gender": "test-gender",
    "group_member_of": "test-group-member-of",
    "has_claim": "test-has-claim",
    "has_credential": "test-has-credential",
    "national_id_number": "test-national-id-number",
    "place_of_birth": "test-place-of-birth"
  }
}